enclaver/src/bin/enclaver/main.rs

```diff
@@ -55,6 +96,52 @@ async fn run(args: Cli) -> Result<()> {
 
             Ok(())
         }
+
+        // Run an enclaver image.
+        Commands::Run {
+            manifest_file,
+            image_name,
+            port_forwards,
+        } => {
+            let image_name = match (manifest_file, image_name) {
+                // If an image was specified, use it
+                (None, Some(image_name)) => Ok(image_name),
+
+                // If no image was specified, either use the specified manifest file or the default
+                // to try to look up the target image name.
+                (manifest_file, None) => {
+                    let manifest_file = manifest_file.unwrap_or_else(|| MANIFEST_FILE_NAME.to_string());
+                    let manifest = load_manifest(manifest_file).await?;
+                    Ok(manifest.target)
+                }
+
+                // Specifying both is an error
+                (Some(_), Some(_)) => Err(anyhow!(
+                    "both an image name and a manifest file were specified"
+                )),
+            }?;
+
+            let mut runner = RunWrapper::new()?;
+
+            let shutdown_signal = enclaver::utils::register_shutdown_signal_handler().await?;
+
+            tokio::select! {
+                res = runner.run_enclaver_image(&image_name, port_forwards) => {
```

Comment:
If `shutdown_signal` comes,the `run_enclaver_image` will get aborted at some inner `.await`. However you don't know which one so `cleanup` now has to account for every case.